src/Diagnostics.RuntimeHost/Controllers/ApiManagement/ApiManagementController.cs

```diff
@@ -12,6 +13,14 @@ namespace Diagnostics.RuntimeHost.Controllers
     [Route(UriElements.ApiManagementServiceResource)]
     public sealed class ApiManagementController : DiagnosticControllerBase<ApiManagementService>
     {
+        protected override ArmResourceType ResourceType
```

Comment:
Look at my suggestion above. If we go with that approach, you dont have to define this override in the Controller class.
The Resource Object will have fields.